tests/MongoDB.EntityFrameworkCore.FunctionalTests/Update/UpdateEntityTests.cs

```diff
@@ -106,6 +120,52 @@ public void Update_realistic_entity()
         }
     }
 
+    [Fact]
+    public void Update_only_updates_modified_fields()
+    {
+        var collection = _tempDatabase.CreateTemporaryCollection<RealisticEntity>();
+
+        var session2 = Guid.NewGuid();
+
+        {
+            using var db = SingleEntityDbContext.Create(collection);
+            db.Entities.Add(new RealisticEntity
+            {
+                _id = ObjectId.GenerateNewId(),
+                session = Guid.NewGuid(),
+                lastCount = 1,
+                lastModified = DateTime.UtcNow.Subtract(TimeSpan.FromDays(5))
+            });
+            db.SaveChanges();
+        }
+
+        // Cause two updates to happen interleaved
+        {
+            using var db1 = SingleEntityDbContext.Create(collection);
+            var entity1 = db1.Entities.First();
+            entity1.lastCount++;
+
+            using var db2 = SingleEntityDbContext.Create(collection);
+            var entity2 = db2.Entities.First();
+            entity2.session = session2;
```

Comment:
Is same value assignment considered as modification (`entity2.session = entity2.session`)?
Whatever the answer is, is it important to capture this behavior in test?

Reply:
This creates two copies of the same entity and sets the properties to different values. There's no same-assignment path here.

Reply:
Yup, the question is whether the same value assignment is considered modification, and whether we should capture the existing behavior in a testcase. For example:

```
entity1.lastCount++;
entity1.session = entity1.session;

entity2.session = session2;

db2.SaveChanges();
db1.SaveChanges(); // Will this overwrite session to  entity1.session ?
```

Reply:
EF shouldn't consider that a change so we don't have test coverage for it as it's out of scope for what our driver actually sees.